types.d.ts

```diff
@@ -1,134 +1,132 @@
+  asyncTimer<T>(func: (...args: any[]) => Promise<T>, stat: string | string[], sampleRate?: number, callback?: StatsCb): (...args: any[]) => Promise<T>;
+  asyncTimer<T>(func: (...args: any[]) => Promise<T>, stat: string | string[], tags?: Tags, callback?: StatsCb): (...args: any[]) => Promise<T>;
+  asyncTimer<T>(func: (...args: any[]) => Promise<T>, stat: string | string[], sampleRate?: number, tags?: Tags, callback?: StatsCb): (...args: any[]) => Promise<T>;
+
+  histogram(stat: string | string[], value: number, callback?: StatsCb): void;
+  histogram(stat: string | string[], value: number, tags?: Tags, callback?: StatsCb): void;
+  histogram(stat: string | string[], value: number, sampleRate?: number, callback?: StatsCb): void;
+  histogram(stat: string | string[], value: number, sampleRate?: number, tags?: Tags, callback?: StatsCb): void;
+
+  distribution(stat: string | string[], value: number, callback?: StatsCb): void;
+  distribution(stat: string | string[], value: number, tags?: Tags, callback?: StatsCb): void;
+  distribution(stat: string | string[], value: number, sampleRate?: number, callback?: StatsCb): void;
+  distribution(stat: string | string[], value: number, sampleRate?: number, tags?: Tags, callback?: StatsCb): void;
+
+  gauge(stat: string | string[], value: number, callback?: StatsCb): void;
+  gauge(stat: string | string[], value: number, tags?: Tags, callback?: StatsCb): void;
+  gauge(stat: string | string[], value: number, sampleRate?: number, callback?: StatsCb): void;
+  gauge(stat: string | string[], value: number, sampleRate?: number, tags?: Tags, callback?: StatsCb): void;
+
+  set(stat: string | string[], value: number, callback?: StatsCb): void;
+  set(stat: string | string[], value: number, tags?: Tags, callback?: StatsCb): void;
+  set(stat: string | string[], value: number, sampleRate?: number, callback?: StatsCb): void;
+  set(stat: string | string[], value: number, sampleRate?: number, tags?: Tags, callback?: StatsCb): void;
+
+  unique(stat: string | string[], value: number, callback?: StatsCb): void;
+  unique(stat: string | string[], value: number, tags?: Tags, callback?: StatsCb): void;
+  unique(stat: string | string[], value: number, sampleRate?: number, callback?: StatsCb): void;
+  unique(stat: string | string[], value: number, sampleRate?: number, tags?: Tags, callback?: StatsCb): void;
+
+  close(callback: (error?: Error) => void): void;
+
+  event(title: string, text?: string, options?: EventOptions, callback?: StatsCb): void;
+  event(title: string, text?: string, options?: EventOptions, tags?: Tags, callback?: StatsCb): void;
+
+  check(name: string, status: DatadogChecksValues, options?: CheckOptions, tags?: Tags, callback?: StatsCb): void;
+
+  public CHECKS: DatadogChecks;
+  public mockBuffer?: string[];
+  public socket: dgram.Socket;
 }
 
-declare const StatsDClient: StatsD;
-export default StatsDClient;
+export default StatsD;
```

Comment:
The default exports seems incorrect before and is incorrect now - if someone imports `import StatsD from '@binaris/hot-shots'` runtime errors will follow (since there's no `exports.default`).

I think that to express `require('@binaris/hot-shots')` returns the StatsD constructor it's possible to use `export = StatsD`, but I'm not sure if it plays nice with the exported class and just removing the default export should be safer.

Reply:
After talking F2F I see you're correct: there is no way of supporting _both_ `export = ...` _and_ new-style exports in the same file.  [TypeScript manual] says:
> Default exports are meant to act as a replacement for this behavior; however, the two are incompatible. TypeScript supports `export =` to model the traditional CommonJS and AMD workflow.

It doesn't work now and it never worked before.  I continue to `export default` so you'll be able to say
```ts
import * as sd from 'hot-shots';
const client = new sd.StatsD();
```
(which seems to work).  WDYT?

Reply:
`import * as sd from 'hot-shots';` works in any case.

I have a suspicion that's is possible to make the export somehow, e.g. in Express (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express/index.d.ts) it's possible to both have
`import * as express from 'express';` and `import { Router } from 'express';`

It can't be copied directly since express is a function and StatsD a constructor, but something like

```typescript
namespace e {

  class StatsD {
    static StatsD: typeof StatsD;
  }
}
declare interface e {
  new(options?: e.ClientOptions): e.StatsD;
}

export = e;
```
should have worked, but doesn't 😞